packages/react-next/src/components/ComboBox/ComboBox.tsx

```diff
@@ -1012,23 +1064,18 @@ export class ComboBox extends React.Component<IComboBoxProps, IComboBoxState> {
         changedOptions[index] = option;
 
         // Call onChange after state is updated
-        this.setState(
-          {
-            selectedIndices: selectedIndices,
-            currentOptions: changedOptions,
-          },
-          () => {
-            // If ComboBox value is changed, revert preview first
-            if (this._hasPendingValue && onPendingValueChanged) {
-              onPendingValueChanged();
-              this._hasPendingValue = false;
-            }
-
-            if (onChange) {
-              onChange(submitPendingValueEvent, option, index, undefined);
-            }
-          },
-        );
+        this.props.hoisted.setSelectedIndices(selectedIndices);
+        this.props.hoisted.setCurrentOptions(changedOptions);
+
+        // If ComboBox value is changed, revert preview first
+        if (this._hasPendingValue && onPendingValueChanged) {
```

Comment:
Similar to Dropdown this is potentially a behavior change since you're now calling onPendingValueChanged and onChange immediately after the setters, instead of in the post-`setState` callback.